From bab29e5b1d21811e6ff617fd4debdce3b81f1d14 Mon Sep 17 00:00:00 2001
From: Tony Cheng <tony.cheng@amd.com>
Date: Thu, 25 Aug 2016 23:00:24 -0400
Subject: [PATCH 0941/1722] drm/amd/dal: remove dead vbios code

- power_up in DAL2 was a "software" initialize which we don't really need
for everything component

Change-Id: Iebb8c3a033e44050da4cf765dd514054d24b8fad
Signed-off-by: Jordan Lazare <Jordan.Lazare@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c           | 17 +++++++++++++++--
 drivers/gpu/drm/amd/dal/dc/dc_bios_types.h              |  3 ---
 drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c |  1 -
 3 files changed, 15 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c b/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c
index 341f42d..e1bf8c3 100644
--- a/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c
+++ b/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c
@@ -4344,13 +4344,26 @@ void dc_bios_destroy_integrated_info(struct dc_bios *dcb,
 	}
 }
 
+static void bios_parser_destroy_integrated_info(
+        struct dc_bios *dcb,
+        struct integrated_info **info)
+{
+        if (info == NULL) {
+                ASSERT_CRITICAL(0);
+                return;
+        }
+
+        if (*info != NULL) {
+                dm_free(*info);
+                *info = NULL;
+        }
+}
+
 /******************************************************************************/
 
 static const struct dc_vbios_funcs vbios_funcs = {
         .get_connectors_number = bios_parser_get_connectors_number,
 
-        .power_up = bios_parser_power_up,
-
         .get_encoders_number = bios_parser_get_encoders_number,
 
         .get_oem_ddc_lines_number = bios_parser_get_oem_ddc_lines_number,
diff --git a/drivers/gpu/drm/amd/dal/dc/dc_bios_types.h b/drivers/gpu/drm/amd/dal/dc/dc_bios_types.h
index 2b370ff..2bc868d 100644
--- a/drivers/gpu/drm/amd/dal/dc/dc_bios_types.h
+++ b/drivers/gpu/drm/amd/dal/dc/dc_bios_types.h
@@ -40,9 +40,6 @@
 
 struct dc_vbios_funcs {
 	uint8_t (*get_connectors_number)(struct dc_bios *bios);
-
-	void (*power_up)(struct dc_bios *bios);
-
 	uint8_t (*get_encoders_number)(struct dc_bios *bios);
 	uint32_t (*get_oem_ddc_lines_number)(struct dc_bios *bios);
 
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
index cca84dd..c434906 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
@@ -1796,7 +1796,6 @@ static void init_hw(struct core_dc *dc)
 	}
 
 	dc->hwss.clock_gating_power_up(dc->ctx, false);
-	dc_bios_power_up(bp);
 	/***************************************/
 
 	for (i = 0; i < dc->link_count; i++) {
-- 
2.7.4

